### PR TITLE
Refactor - Simplifies `emit_shift()`

### PR DESCRIPTION
**Pull request by Lichtso**

### CODE REVIEW

File: src/jit.rs [hunk 1]
```diff
@@ -1257,34 +1257,20 @@ impl<'a, C: ContextObject> JitCompiler<'a, C> {
...
-            } else {
-                self.emit_ins(X86Instruction::alu_immediate(size, 0xc1, opcode_extension, destination, immediate, None));
-                return;
-            }
+            self.emit_ins(X86Instruction::alu_immediate(size, 0xc1, opcode_extension, destination, immediate, None));
+            return;
         }
         if let OperandSize::S32 = size {
-            self.emit_ins(X86Instruction::alu_immediate(OperandSize::S32, 0x81, 4, destination, -1, None)); // Mask to 32 bit
+            self.emit_ins(X86Instruction::mov(OperandSize::S32, destination, destination)); // Mask to 32 bit
```
Comment (LucasSte) [inline src/jit.rs]: This is not a mask anymore (see the comment next to the instruction)

File: src/jit.rs [hunk 2]
```diff
@@ -1257,34 +1257,20 @@ impl<'a, C: ContextObject> JitCompiler<'a, C> {
...
         }
         if source == RCX {
-            if destination == RCX {
-                self.emit_ins(X86Instruction::alu_immediate(size, 0xd3, opcode_extension, destination, 0, None));
-            } else {
-                self.emit_ins(X86Instruction::push(RCX, None));
-                self.emit_ins(X86Instruction::alu_immediate(size, 0xd3, opcode_extension, destination, 0, None));
-                self.emit_ins(X86Instruction::pop(RCX));
-            }
```
Comment (LucasSte) [inline src/jit.rs:1277]: Can't the destination still be a register other than RCX?

Comment (Lichtso) [reply src/jit.rs:1277]: yep, but there is no need to save and restore `RCX` when it is a source

File: src/jit.rs (hunk 1 above)
Comment (Lichtso) [reply src/jit.rs]: I renamed it from "mask" to "truncate"

**Review by LucasSte — APPROVED** [review]

**Pull request merged**